MovingBlocks/Terasology
# fix: Time out of sync for players in multiplayer

### Contains

Fix proposal for Issue #3670 .
Most of the code comes from @godfather2 's pull request #3890 , closed on 29 August 2020, which attempted to fix time synchronisation when connecting with another client, and when changing time in Debug mode.

Using setWorldTime command on either client should now update time on both clients.


### How to test

- Start a local multiplayer game
- Start up another client
- Wait 5 minutes or so before connecting.
-> The second client should now have its time (almost) synchronized with the first one (see next section).
- Change time on either client, using arrow keys in Debug mode.
-> Both clients should now have their times updated.
- Use `setWorldTime 0.9` on either client.
-> Time should now be set to 0.9 on both clients.

### Outstanding before merging

As opposed to the description of the issue, time is already (mostly) synchronized when connecting to a multiplayer game.
As explained in PR #3890 , the host's time is sent at the beginning of the connection process, which causes a delay of about 0.03 day (or more, depending on the capability of the computer used) when the new client eventually connects.
